.gitignore

```diff
@@ -1,6 +1,14 @@
 .DS_Store
-.classpath
-.project
-.settings
-target
 test-output
+
+# Eclipse #
+/.classpath/
+/.project/
+/.settings/
+
+# Maven #
+*/target/
+
+# IntelliJ #
+/.idea/
+*.iml
```

Comment:
If you have a multi-module project, each sub-module has an *.iml file at it's root